### ISSUE TITLE
Response of InitializeHealthKit always returns 1. I also included the line that causes this bug.

### ISSUE THREAD

**Issue by furkannarin** [agencyenterprise/react-native-health#354]
**Describe the bug**
initHealthKit function always returns 1 as a response value.

**To Reproduce**
Use initHealthKit function. Its response will always be 1 no matter the permission is given or not.

**Expected behavior**
Function should return 0 if permission is not granted.

**Smartphone (please complete the following information):**
- Device: iPhone 15
- OS: ios 17
- Version 0.19.1

**Additional context**
Inside **RCTAppleHealthKit/RCTAppleHealthKit.m** file, line 667:
`callback(@[[NSNull null], @true]);`

This line always returns true. Instead, it should return the success value from the completion function. I've tried to change it but I am no Objective-C expert so I failed. Looks like the callback block has its own limitations??

**TLDR**: line 667 @true should be the **_BOOL success_** variable

```
 [self.healthStore requestAuthorizationToShareTypes:writeDataTypes readTypes:readDataTypes completion:^(BOOL success, NSError *error) {
            if (!success) {
                NSString *errMsg = [NSString stringWithFormat:@"Error with HealthKit authorization: %@", error];
                 NSLog(@"%@", errMsg);
                callback(@[RCTMakeError(errMsg, nil, nil)]);
                return;
            } else {
                dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
                    callback(@[[NSNull null], @true]);
                });
            }
        }];
    } else {
        callback(@[RCTMakeError(@"HealthKit data is not available", nil, nil)]);
    }
```

**Comment by kfchan-git** [comment]
i'm also having the same issue

**Comment by utkarshgupta16** [comment]
Same issue, and also Error variable is also always null no matter whether you have given permission or not.

**Comment by olarra** [comment]
Could this bug be opened again? The bug is still present. permissions is always 1

```javascript
"permissions": {
    "read": [
      1
    ],
    "write": []
  }
```
it should be returning 2 for Authorized

```javascript
  export enum HealthStatusCode {
    NotDetermined = 0,
    SharingDenied = 1,
    SharingAuthorized = 2,
  }

**Comment by kfchan-git** [comment]
i used to stamp to determine whether if the permission is on or not; 
etc, if there's data last week (>= 7 days) and no data within this week = permission off

**Comment by Snouzy** [comment]
the bug is still here 😅
